ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -226,9 +230,10 @@ public int persistColumnStats(Hive db, Table tbl) throws HiveException, MetaExce
         if (!(tbl.isMaterializedView() || tbl.isView() || tbl.isTemporary())) {
           setOrRemoveColumnStatsAccurateProperty(db, tbl, colStatDesc.getColName(), success);
         }
+      } else {
```

Comment:
I can see this is fixed in [HIVE-27421](https://issues.apache.org/jira/browse/HIVE-27421)), and PR is told to be ready to merge, don't we want to fix this in that PR?

Reply:
I wish it would be ready to merge, tests are failing. btw, I've reopened it, but I can't push amends to it

Reply:
btw, that PR has some weird logic, it calls `setColStatistics` twice
````
tbl.getStorageHandler().setColStatistics(tbl, colStats)
````